recipes/newrelic/apm/php/debian.yml

```diff
@@ -296,30 +309,85 @@ install:
+              sed_slash_ini=$(echo "${ini}"  | sed 's/\//\\\//g')
+              php_bin=$(sed '1,/final pi_inidir_.*='"${sed_slash_ini}"'/d;/php -i output follows/,$d' $NR_INSTALL_LOG | sed -n 's/.*pi_bin=\(.*\/\)/\1/p')
+              if [ -z "${php_bin}" ]; then
+              #
+              # When there is only one PHP detected, the info format is different,
+              # we could detect from the log file, but since there is only one,
+              # just use `php`.
+              #
+                php_bin="php"
+              fi
+
+              #
+              # Modify the fresh copy of a brand new New Relic INI file.
+              #
+              ini_full_name="${ini}/newrelic.ini"
+              if [ ! -f "$ini_full_name" ]; then
+                continue
+              fi
+              sed -i "s/newrelic.appname = \"PHP Application\"/newrelic.appname = \"{{.NR_PHP_APPLICATION}}\"/" $ini_full_name
+              if [ "{{.NEW_RELIC_REGION}}" = "STAGING" ]; then
+                sed -i 's/;newrelic.daemon.collector_host = ""/newrelic.daemon.collector_host = "staging-collector.newrelic.com"/' $ini_full_name
+                sed -i 's/;newrelic.loglevel = "info"/newrelic.loglevel = "verbosedebug"/' $ini_full_name
+                sed -i 's/;newrelic.daemon.loglevel = "info"/newrelic.daemon.loglevel = "debug"/' $ini_full_name
+              fi
+
+              #
+              # Determine the target directory that the symlink is pointing to.
+              #
+              target_ini_dir="/etc/php5/conf.d/"
+              if [ -x /usr/sbin/phpquery ]; then
+                target_ini_dir="${ini}/../../mods-available/"
+              elif [ -x /usr/sbin/php5endmod ]; then
+                target_ini_dir="/etc/php5/mods-available/"
+              fi
+              if [ ! -d $target_ini_dir ]; then
+                echo -e "{{.RED}}Warning: Unable to move ${ini_full_name} to ${target_ini_dir}{{.GRAY}}"
+                continue
+              fi
+              #
+              # Move newrelic.ini created by tarball to the name that the installer will use.
+              # This ensures we always get the most up to date newrelic.ini file on the
+              # system.
+              #
+              mv ${ini_full_name} ${target_ini_dir}
+
+              #
+              # We rely on the package installer to enable the newrelic module.
+              #
+
+              #
+              # Three important pieces of information: the php binary location, the php ini
+              # file location, the actual (not symlink) New Relic ini file location.
+              # If we have web directories, copy the three key bits of info to the web file;
+              # otherwise copy to the cli file.  We will save the php directory, the target
+              # directory where the ini file is, and the php bin directory.
+              #
+              if [ -z "${CLI_DIRS}" ]; then
```

Comment:
I only see this being set after this if clause and it is only referenced in these two places. Is there something else that sets or references this not in the recipe?

Reply:
The first set of dirs the loop goes through is web, after it is done with that, the next set of dirs will be CLI so it is marked.

Reply:
Could there be multiple web ini dirs if there are multiple versions of PHP installed on the system? If so, do we risk setting the cli flag too soon?

Reply:
All the web dirs are bundled together in an array, all the cli dirs are bundled together in an array.
After we are done with the web dirs array, we set the flag and move on to handling the cli dirs array.

Reply:
I see that now. Thank you for so patiently pointing out the parts of this code I missed!